### DISCUSSION TITLE
Expose the `builtins` module as a public API

### DISCUSSION THREAD

**Discussion by jedel1043** [boa-dev/boa#1692] [Ideas]
This should make it possible to call Javascript built-in functions from Rust, enabling some use cases like making a Javascript script within Rust without using any Javascript code, which sounds promising.

**Comment by lastmjs** [comment]
I would really like this change!

**Reply by jasonwilliams** [reply]
@lastmjs could you give some detail what you would use it for?

**Reply by lastmjs** [reply]
I had a somewhat lengthy discussion on what I'm trying to do in the discord: https://discord.com/channels/595323158140158003/595382036215365633/908424507285459025

Basically I am working on a Node/Deno for the Internet Computer, a decentralized BFT general purpose cloud (a blockchain computer). See more about the Internet Computer here: https://dfinity.org/

The project I'm working on is called Azle: https://github.com/lastmjs/azle

It's a TypeScript CDK (Canister Development Kit). Canisters are like containers, but deployed to the Internet Computer instead of a cloud provider like AWS.

The Internet Computer uses wasmtime as its core virtual machine. It provides Wasm imports to access the Internet Computer environment. I'm using boa within a Rust canister to bridge TypeScript/JavaScript and the Internet Computer.

I've run into the issue of exposing Internet Computer APIs to the JavaScript environment. I need to get data structures retrieved from the IC into JsValue's. For example, I can call a `raw_rand` API in Rust to get randomness generated from the IC as a `Vec<u8>`. I want to turn that into a Uint8Array and pass it into the JavaScript environment when in JS the developer calls a `rawRand` function.

I'm having a bit of trouble doing this with the builtins not exposed.

Basically, I want complete flexibility to send data from boa -> Rust and from Rust -> boa. Azle is most likely going to become very popular on the Internet Computer soon, so expect a lot of usage of boa (I hope it can handle it).

**Comment by lastmjs** [comment]
Could we get these exposed? I'm happy to do a PR.
